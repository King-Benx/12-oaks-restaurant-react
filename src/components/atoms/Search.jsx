import PropTypes from "prop-types";
import { useState } from "react";

function Search({ onSubmitHandler }) {
  const [state, setState] = useState({location:"", term:""});
  const onSubmit = (e) => {
    e.preventDefault();
    onSubmitHandler(state);
};
  return (
    <div className="container mx-auto bg-gradient-to-r from-teal-400 via-teal-500 to-teal-600 my-10 py-10 rounded-lg">
      <form onSubmit={(e) => onSubmit(e)} className="flex justify-center">
        <div className="w-full grid gap-6 mb-6 md:grid-cols-3 px-6">
          <div>
            <input
              type="text"
              id="first_name"
              className="bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg focus:ring-blue-500 focus:border-blue-500 block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500"
              placeholder="Location"
              required
              value={state.location}
              onChange={e =>
                setState(prevState=>({
                  ...prevState,
                  location: e.target.value,
                }))
              }
            />
          </div>
          <div>
            <input
              type="text"
              id="last_name"
              className="bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg focus:ring-blue-500 focus:border-blue-500 block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500"
              placeholder="Term"
              required
              value={state.term}
              onChange={e =>
                setState(prevState=>({
                  ...prevState,
                  term: e.target.value,
                }))
              }
            />
          </div>

          <button
            type="submit"
            className="w-10 h-10 p-2.5 text-white bg-gradient-to-r from-cyan-500 via-cyan-600 to-cyan-700 hover:bg-gradient-to-br focus:ring-4 focus:outline-none focus:ring-cyan-300 dark:focus:ring-cyan-800 font-medium rounded-full text-sm text-center"
          >
            <svg
              aria-hidden="true"
              className="w-5 h-5"
              fill="currentColor"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                fillRule="evenodd"
                d="M10.293 3.293a1 1 0 011.414 0l6 6a1 1 0 010 1.414l-6 6a1 1 0 01-1.414-1.414L14.586 11H3a1 1 0 110-2h11.586l-4.293-4.293a1 1 0 010-1.414z"
                clipRule="evenodd"
              ></path>
            </svg>
            <span className="sr-only">Search Icon</span>
          </button>
        </div>
      </form>
    </div>
  );
}

Search.propTypes = {
  onSubmitHandler: PropTypes.func.isRequired,
};

export default Search;
