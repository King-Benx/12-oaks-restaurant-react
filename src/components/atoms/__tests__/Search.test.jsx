import {render, screen, fireEvent, cleanup} from "@testing-library/react";
import renderer from "react-test-renderer";
import Search from "../Search";
import "@testing-library/jest-dom";

afterEach(()=> cleanup());

test("should render the Search component", ()=>{
    render(<Search onSubmitHandler={()=>{}}/>);
    const locationInput = screen.getByLabelText("location");
    const termInput = screen.getByLabelText("term");
    const buttonSubmit = screen.getByLabelText("submit");
    const searchElement = screen.getByTestId("search");
    fireEvent.change(locationInput,{target:{value:"london"}});
    fireEvent.change(termInput,{target:{value:"restaurant"}});
    fireEvent.click(buttonSubmit);
    expect(searchElement).toBeInTheDocument();
    expect(searchElement).toHaveTextContent("Search Icon");
    expect(locationInput.value).toBe("london");
    expect(termInput.value).toBe("restaurant");
});

test("matches snapshot", () => {
    const tree = renderer.create(<Search onSubmitHandler={()=>{}}/>).toJSON();
    expect(tree).toMatchSnapshot();
});
