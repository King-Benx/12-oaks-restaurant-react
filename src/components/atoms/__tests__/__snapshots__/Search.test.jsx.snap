// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="container mx-auto bg-gradient-to-r from-teal-400 via-teal-500 to-teal-600 my-10 py-10 rounded-lg"
  data-testid="search"
>
  <form
    className="flex justify-center"
    onSubmit={[Function]}
  >
    <div
      className="w-full grid gap-6 mb-6 md:grid-cols-3 px-6"
    >
      <div>
        <input
          aria-label="location"
          className="bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg focus:ring-blue-500 focus:border-blue-500 block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500"
          id="first_name"
          onChange={[Function]}
          placeholder="Location"
          required={true}
          type="text"
          value=""
        />
      </div>
      <div>
        <input
          aria-label="term"
          className="bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg focus:ring-blue-500 focus:border-blue-500 block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500"
          id="last_name"
          onChange={[Function]}
          placeholder="Term"
          required={true}
          type="text"
          value=""
        />
      </div>
      <button
        aria-label="submit"
        className="w-10 h-10 p-2.5 text-white bg-gradient-to-r from-cyan-500 via-cyan-600 to-cyan-700 hover:bg-gradient-to-br focus:ring-4 focus:outline-none focus:ring-cyan-300 dark:focus:ring-cyan-800 font-medium rounded-full text-sm text-center"
        type="submit"
      >
        <svg
          aria-hidden="true"
          className="w-5 h-5"
          fill="currentColor"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M10.293 3.293a1 1 0 011.414 0l6 6a1 1 0 010 1.414l-6 6a1 1 0 01-1.414-1.414L14.586 11H3a1 1 0 110-2h11.586l-4.293-4.293a1 1 0 010-1.414z"
            fillRule="evenodd"
          />
        </svg>
        <span
          className="sr-only"
        >
          Search Icon
        </span>
      </button>
    </div>
  </form>
</div>
`;
